src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -49,15 +55,60 @@ public static Drivetrain getInstance() {
-                      SwerveModule backRight, AHRS gyro) {
+                       SwerveModule backRight, AHRS gyro) {
         super(NAMESPACE_NAME);
         this.frontLeft = frontLeft;
         this.frontRight = frontRight;
         this.backLeft = backLeft;
         this.backRight = backRight;
+        stupidPointlessArray = new SwerveModulePosition[] {frontLeft.getPosition(),
+                frontRight.getPosition(), backLeft.getPosition(), backRight.getPosition()};
         kinematics = new SwerveDriveKinematics(FRONT_LEFT_WHEEL_POSITION,
                 FRONT_RIGHT_WHEEL_POSITION, BACK_LEFT_WHEEL_POSITION, BACK_RIGHT_WHEEL_POSITION);
         this.gyro = gyro;
+        odometry = new SwerveDriveOdometry(kinematics, gyro.getRotation2d(), stupidPointlessArray,
+                new Pose2d());
+        RobotConfig config;
+        try{
+            config = RobotConfig.fromGUISettings();
+        } catch (Exception e) {
+            // Handle exception as needed
+            e.printStackTrace();
+            config = null;
+        }
+
+        // Configure AutoBuilder last
+        AutoBuilder.configure(
+                this::getPose2d, // Robot pose supplier
+                this::resetPose2d, // Method to reset odometry (will be called if your auto has a starting pose)
+                this::getRobotRelativeSpeeds, // ChassisSpeeds supplier. MUST BE ROBOT RELATIVE
+                (speeds, feedforwards) -> driveRobotRelative(speeds), // Method that will drive the robot given ROBOT RELATIVE ChassisSpeeds. Also optionally outputs individual module feedforwards
+                new PPHolonomicDriveController( // PPHolonomicController is the built in path following controller for holonomic drive trains
+                        new PIDConstants(0.0, 0.0, 0.0), // Translation PID constants
+                        new PIDConstants(0.01, 0.0, 0.0) // Rotation PID constants
+                ),
+                config, // The robot configuration
+                () -> {
+                    // Boolean supplier that controls when the path will be mirrored for the red alliance
+                    // This will flip the path being followed to the red side of the field.
+                    // THE ORIGIN WILL REMAIN ON THE BLUE SIDE
+
+                    var alliance = DriverStation.getAlliance();
+                    if (alliance.isPresent()) {
+                        return alliance.get() == DriverStation.Alliance.Red;
+                    }
+                    return false;
+                },
+                this // Reference to this subsystem to set requirements
+        );
+    }
+
+    @Override
+    public void periodic() {
+        super.periodic();
+        stupidPointlessArray = new SwerveModulePosition[]{frontLeft.getPosition(),
+                frontRight.getPosition(), backLeft.getPosition(), backRight.getPosition()};
+        odometry.update(gyro.getRotation2d(), stupidPointlessArray);
```

Comment:
instead of using `stupidPointlessArray`, call `new SwerveModulePosition[](...)` directly